Userland/DevTools/HackStudio/main.cpp

```diff
@@ -211,18 +209,18 @@ bool semantic_syntax_highlighting_is_enabled()
 
 }
 
-static ErrorOr<NonnullRefPtr<HackStudioWidget>> create_hack_studio_widget(bool mode_coredump, DeprecatedString const& absolute_path_argument, pid_t pid_to_debug)
+static ErrorOr<NonnullRefPtr<HackStudioWidget>> create_hack_studio_widget(bool mode_coredump, StringView raw_path_argument, pid_t pid_to_debug)
 {
-    auto project_path = Core::DeprecatedFile::real_path_for(".");
-    if (!mode_coredump) {
-        if (!absolute_path_argument.is_null())
-            project_path = absolute_path_argument;
-        else if (auto last_path = last_opened_project_path(); last_path.has_value())
-            project_path = last_path.release_value();
-    }
-
-    if (pid_to_debug != -1)
+    DeprecatedString project_path;
+    if (pid_to_debug != -1 || mode_coredump)
         project_path = "/usr/src/serenity";
+    else if (!raw_path_argument.is_null())
+        // FIXME: Validation is unintentional, and should be removed when migrating to String.
+        project_path = TRY(DeprecatedString::from_utf8(raw_path_argument));
+    else if (auto last_path = last_opened_project_path(); last_path.has_value())
+        project_path = last_path.release_value();
+    else
+        project_path = TRY(FileSystem::real_path("."sv)).to_deprecated_string();
```

Comment:
You could limit the churn on this function in the future by using a String now (we are in an ErrorOr context after all) and just call .to_deprecated_string() in the create() call.

Reply:
... or not :D. We will have the churn after all